validator/checker.py

```diff
@@ -0,0 +1,246 @@
+import re
```

Comment:
Add a docstring here with a link to the exact place (with commit hash) where you copied this from so we have a reference